Prog/vacant-lots-arlington.Rmd

```diff
@@ -180,40 +161,436 @@ First, we look at all the options, and filter out all the observations marked as
+                         ZN_DESIG,
+                         zoning)) %>% 
+  select(-c(OBJECTID, ZONECODE, LABEL, SYMBOL, ZN_DESIG_6, GZDC))
+
+```
+
+Recombine with observations with missing coordinates, check that number of rows is the same.
+
+```{r get-final}
+
+vacanttotal <- vacant1 %>% 
+  filter(is.na(lat) | is.na(long)) %>% 
+  mutate(zoning_bk = zoning) %>% 
+  bind_rows(vacant3)
+
+stopifnot(nrow(vacanttotal) == nrow(vacant1))
+
+```
+
+The last step is to identify a zoning designation for observations that either...
+
+1. Have missing coordinates and have a dual zoning designation from Black Knight
+2. Have a dual zoning designation from Black Knight but one of the zoning codes does not match the results from the spatial join.
+
+First, isolate these observations and separate each zoning code into its own observation. 
+**NOTE:** For Arlington, the max codes was 4- adjust the `into = ` argument accordingly for each jurisdiction.
+
+```{r real-final}
+
+vacantweird <- vacanttotal %>% 
+  filter((str_detect(zoning, "/") == TRUE & 
+            str_detect(zoning_bk, ZN_DESIG) == FALSE) |
+          (str_detect(zoning, "/") == TRUE &
+           missing_coord == TRUE)) %>% 
+  separate(zoning_bk, 
+           into = c("z1", "z2", "z3", "z4"),
+           sep = "/",
+           remove = FALSE)
+
+```
+Get rid of values that are NOT in the cleaned zoning code data. These codes do not allow for residential development.
+
+```{r no-non-res}
+
+vacantweird <- vacantweird %>% 
+  mutate_at(vars(z1, z2, z3, z4),
+            ~ ifelse(!. %in% zoningcode1$zoning_code,
+                     NA,
+                     .))
+
+```
+
+Convert to long so that we can list the zoning variables- then `group_by` the parcel ID, and check the lengths to make sure that is unique, so that we can merge results back in.
+
+```{r long}
+
+vacantlong <- vacantweird %>% 
+  gather(key = "position", value = "code", -c(1:46, 51))
+
+vacantweird2 <- vacantlong %>% 
+  group_by(assessorsparcelnumberapnpin) %>% 
+  summarize(zlist = list(code))
+
+stopifnot(nrow(vacantweird) == nrow(vacantweird2))
+
+```
+
+If there is only one code left after getting rid of the ones that don't allow for residential codes, take that code.
+
+```{r true1}
+vacantweird2 <- vacantweird2 %>% 
+  mutate(count = map(zlist, ~ sum(!is.na(.)))) %>% 
+  mutate(zoning_true = ifelse(count == 1,
+                              map(zlist, max, na.rm = TRUE),
+                              ""))
+```
+
+Priotize codes that are purely residential (begin with R). If there are more than 1 "R" zoning codes, we want to take the one with the lower number (take the lower density code, so we do not overestimate, since part of the building will be in the lower density area). We do this by extracting the numbers from the zoning codes, taking the minimum, and then selecting the element that contains that number.
+
+```{r true2}
+
+vacantweird2 <- vacantweird2 %>% 
```

Comment:
@awunderground would like feedback on this step- it gets the job done, but I feel like there could be a better solution.